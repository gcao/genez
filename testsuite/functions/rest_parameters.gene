# Test rest parameters/varargs

# Basic rest parameter function
(fn collect args
    (println args))

(collect)
(collect 1)
(collect 1 2 3)

# Rest parameter with regular parameters
(fn format [prefix] items
    (println prefix items))

(format "Items:")
(format "Numbers:" 1 2 3)

# Multiple regular parameters with rest
(fn log [level prefix] messages
    (println #"[#{level}] #{prefix}:" messages))

(log "INFO" "Server")
(log "ERROR" "Database" "Connection failed" "Retry in 5s")

# Empty rest parameter behavior
(fn optional_args [required] opts
    (println "Required:" required "Options:" opts))

(optional_args "value")
(optional_args "value" "opt1" "opt2")