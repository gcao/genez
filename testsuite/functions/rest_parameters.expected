[]
[1]
[1 2 3]
Items: []
Numbers: [1 2 3]
[INFO] Server: []
[ERROR] Database: ["Connection failed" "Retry in 5s"]
Required: value Options: []
Required: value Options: ["opt1" "opt2"]